tests_e2e/orchestrator/lib/agent_test_suite.py

```diff
@@ -230,9 +234,9 @@ def _initialize(self, environment: Environment, variables: Dict[str, Any], lisa_
 
         if isinstance(environment.nodes[0], LocalNode):
             # We need to create a new VMSS.
-            # Use the same naming convention as LISA for the scale set name: lisa-<runbook name>-<run id>-e0-n0. Note that we hardcode the resource group
-            # id to "e0" and the scale set name to "n0" since we are creating a single scale set.
-            self._resource_group_name = f"lisa-{self._runbook_name}-{RUN_ID}-e0"
+            # Use the same naming convention as LISA for the scale set name: lisa-<runbook name>-<run id>-e0-n0. Note
+            # that we hardcode the scale set name to "n0" since we are creating a single scale set.
+            self._resource_group_name = f"lisa-{self._runbook_name.lower()}-{RUN_ID}-e{self._vmss_resource_group_count}"
```

Comment:
ipaddress cannot have upercase characters in it's domain or we hit the following error. Making runbook name lowercase here

`Code: InvalidTemplateDeployment
Message: The template deployment 'TestDeployment' is not valid according to the validation procedure. The tracking id is '043fde1e-6cff-43ff-85d6-a15dd8f97690'. See inner errors for details.
Exception Details:	(InvalidDomainNameLabel) The domain name label lisa-WALinuxAgent-20231031-191104-824-e0-n0 is invalid. It must conform to the following regular expression: ^[a-z][a-z0-9-]{1,61}[a-z0-9]$.
	Code: InvalidDomainNameLabel
	Message: The domain name label lisa-WALinuxAgent-20231031-191104-824-e0-n0 is invalid. It must conform to the following regular expression: ^[a-z][a-z0-9-]{1,61}[a-z0-9]$.
	Target: /subscriptions/***/resourceGroups/lisa-WALinuxAgent-20231031-191104-824-e0/providers/Microsoft.Network/publicIPAddresses/e2e-testPublicLbIp
`

Reply:
thanks. this is worth adding as a comment in the code